### ISSUE TITLE
glfwCreateStandardCursor should return errors

### ISSUE THREAD

**Issue by OhadRau** [bryphe/reason-glfw#73]
`glfwCreateStandardCursor` can sometimes return errors, but since these aren't handled in the Reason code this will cause a lot of confusion for the user. There's a number of ways to approach this--the function can return an `option`/error type, it could throw an exception, or do something else entirely. One example of a place this can happen is if `glfwCreateStandardCursor` is called before `glfwInit`, in which case the cursor inexplicably won't change when `glfwSetCursor` gets called with that value. Since the most natural way to use the library will often lead to that case, it's very important to handle this in some way that immediately comes to the user's attention (rather than just kicking the can down the road for them to figure out later).

**Comment by OhadRau** [comment]
My personal suggestion is exceptions because they do a good job of uncoupling the creation code from the error recovery code. For example, if we returned an option and we tried to make a top-level value using this function, we might end up with something like this:

```
let arrow =
  switch(glfwCreateStandardCursor(GLFW_ARROW_CURSOR)) {
  | Some(cursor) => cursor
  | None => glfwCreateStandardCursor(GLFW_ARROW_CURSOR)
  /* ... What else can we do with None here? */
  };
```

which wouldn't work because it doesn't wait on `glfwInit` to get called. In my opinion this would overcomplicate dealing with this library but wouldn't significantly improve the errors we get. However, since OCaml uses unchecked exceptions, the user doesn't have to try handling this at all (the cases where errors are thrown are quite exceptional, and generally won't happen during the normal runtime of the program). Instead, they'll just see something in `stderr` along the lines of "Error: Couldn't create a glfwCursor, as glfw was not yet initialized" and immediately know what to do.

**Comment by bryphe** [comment]
Looks like this was fixed with #81
